### PR TITLE
Update train_util.py to allow for using SGD without momentum

### PR DESCRIPTION
**Pull request by ddpasa**
Using momentum increases VRAM requirement by quite a bit, since you have to keep track of past gradients.  momentum=0 might not be optimal, but is nice to have if you're using low-memory GPUs.

### CODE REVIEW

**Comment by FurkanGozukara** [comment]
can you give some information regarding what is SGD  and momentum?

**Comment by ddpasa** [comment]
SGD is stocastic gradient descent optimizer here: 
https://en.wikipedia.org/wiki/Stochastic_gradient_descent
https://pytorch.org/docs/stable/generated/torch.optim.SGD.html

Momentum uses past gradients to improve convergence.  It works nice, but also increases VRAM usage quite a bit.  momentum=0 uses less VRAM, allowing training on older GPUs.

**Comment by FurkanGozukara** [comment]
@ddpasa thanks a lot for explanation

**Comment by ddpasa** [comment]
@kohya-ss what do you think?  Is this okay to merge?

**Comment by ddpasa** [comment]
It's kind of funny to call it SGDNesterov because the underlying optimizer used is just pytorch SGD, which supports SGD with or without Nesterov momentum.  The most ideal solution is to rename it to just SGD, but that would probably break too many things.
